### PR TITLE
[DataSync] Add throwOnConflict strategy and check for validity for conflictConfig

### LINKED ISSUE #1930: Delta - missing validation for invalid conflict key
We enable conflict per model and not operation. If someone will pass operation name should get error.

### PR DESCRIPTION
**Pull request by ssd71**
Fixes #1930 #1914

### CODE REVIEW

File: packages/graphback-datasync/src/DataSyncPlugin.ts [hunk 1]
```diff
@@ -59,6 +61,13 @@ export class DataSyncPlugin extends GraphbackPlugin {
       console.warn("No DataSync Models detected, ensure that your models are properly annotated.")
     }
 
+    if (modelConflictConfigSet.size !== 0) {
+      console.log("The following models from conflictConfig could not be found, consider adding them to the schema and/or adding the @datasync annotation if you have not done so: ");
```
Comment (craicoverflow) [inline packages/graphback-datasync/src/DataSyncPlugin.ts:65]:
```suggestion
    if (modelConflictConfigSet.size !== 0) {
      console.log("The following models from conflictConfig could not be found, consider adding them to the schema and/or adding the @datasync annotation if you have not done so.");
```

File: docs/datasync/conflict-resolution-strategies.md [hunk 2]
```diff
@@ -49,6 +49,11 @@ This strategy ensures that updates always resolve to whatever the client had sen
 This strategy ensures that in the event of a update conflict, the client's update will never overwrite any field that has changed since the client last fetched it. If the object has been deleted in the database, the client will be notified of it by way of a `ConflictError`. For delete conflicts, the client is informed of the conflict via a `ConflictError`.
 
 
+### ThrowOnConflict
+
+This strategy throws a `ConflictError` every time a conflict occurs, so that the client can be informed about the conflicts and can possibly resolve the conflicts.
```
Comment (wtrocki) [inline docs/datasync/conflict-resolution-strategies.md:54]:
```suggestion
This strategy throws a `ConflictError` every time a version field from the client is different from than server-side version. The client can be informed about potential conflict and can resolve it if needed.
```

File: packages/graphback-datasync/src/DataSyncPlugin.ts [hunk 3]
```diff
@@ -59,6 +61,13 @@ export class DataSyncPlugin extends GraphbackPlugin {
...
+      modelConflictConfigSet.forEach((modelName: string) => {
+        console.log(modelName)
```
Comment (wtrocki) [inline packages/graphback-datasync/src/DataSyncPlugin.ts:67]: Remove

File: packages/graphql-serve/src/runtime.ts [hunk 4]
```diff
@@ -5,11 +5,17 @@ import { MongoMemoryServer } from 'mongodb-memory-server';
 import { MongoClient, Db } from 'mongodb';
 import { loadModel } from './loadModel';
 import { GraphQLSchema } from 'graphql';
-import { createDataSyncAPI, ClientSideWins, ServerSideWins, ConflictResolutionStrategy } from "@graphback/datasync"
+import { createDataSyncAPI, ClientSideWins, ServerSideWins, ConflictResolutionStrategy, ThrowOnConflict } from "@graphback/datasync"
 import { IResolverObject } from 'apollo-server-express';
 
 
-export type ConflictResolutionStrategyName = "clientSideWins"|"serverSideWins";
+export type ConflictResolutionStrategyName = keyof typeof ConflictStrategyMap;
```
Comment (wtrocki) [inline packages/graphql-serve/src/runtime.ts:12]: Comments please

File: docs/datasync/conflict-resolution-strategies.md (hunk 2 above)
Comment (ssd71) [reply docs/datasync/conflict-resolution-strategies.md:54]: This strategy currently only throws if there is conflict i.e. server and client change same fields, should I change it?

File: packages/graphback-datasync/src/DataSyncPlugin.ts (hunk 3 above)
Comment (ssd71) [reply packages/graphback-datasync/src/DataSyncPlugin.ts:67]: This is the check for invalid conflict key #1930, should I remove it?

File: packages/graphback-datasync/src/util.ts [hunk 5]
```diff
@@ -158,3 +158,11 @@ export const ClientSideWins: ConflictResolutionStrategy = {
   }
 }
 
+export const ThrowOnConflict: ConflictResolutionStrategy = {
+  resolveUpdate(conflict: ConflictMetadata): any {
```
Comment (wtrocki) [inline packages/graphback-datasync/src/util.ts:162]: Actually - would merge happen in this case?

File: docs/datasync/conflict-resolution-strategies.md (hunk 2 above)
Comment (wtrocki) [reply docs/datasync/conflict-resolution-strategies.md:54]: Actually this is wrong. Our implementation will do merge.

Comment (wtrocki) [reply docs/datasync/conflict-resolution-strategies.md:54]: Yep. Still if you can rephrase it a little bit

File: packages/graphback-datasync/src/DataSyncPlugin.ts (hunk 3 above)
Comment (wtrocki) [reply packages/graphback-datasync/src/DataSyncPlugin.ts:67]: Sorry for not being clear. 

Having single console log is fine.

${modelConflictConfigSet.join("\n")}

**Review by wtrocki — APPROVED** [review]
Works like dream!

**Comment by wtrocki** [comment]
Conflicts to be resolved.

File: packages/graphback-datasync/src/DataSyncPlugin.ts [hunk 6]
```diff
@@ -40,12 +40,14 @@ export class DataSyncPlugin extends GraphbackPlugin {
 
       return schema;
     }
+    const modelConflictConfigSet = new Set(Object.keys(this.config.conflictConfig.models))
```
Comment (machi1990) [inline packages/graphback-datasync/src/DataSyncPlugin.ts:43]: Does it need to be a `Set`

Comment (ssd71) [reply packages/graphback-datasync/src/DataSyncPlugin.ts:43]: I think if it was an array, we would have to search for the model name every time this loop runs

**Pull request merged**